packages/subgraph/src/mappingHelpers.ts

```diff
@@ -523,10 +523,14 @@ function updateATSBalanceAndUpdatedAt(
     let superTokenContract = SuperToken.bind(
         Address.fromString(accountTokenSnapshot.token)
     );
-    let newBalance = superTokenContract.balanceOf(
-        Address.fromString(accountTokenSnapshot.account)
+    let newBalanceResult = superTokenContract.try_realtimeBalanceOf(
```

Comment:
why being defensive here?

is there anyway to have warning logs in subgraph, and being able to monitor them?

Reply:
If we don't do the try, the whole subgraph breaks, but probably a good idea to put warning logs in the event that it doesn't work, not sure why it would revert though

Reply:
Main change is also from balance of to realtime balance of

Reply:
If it doesn't work, it should break, no?

Reply:
Probably not? We're pretty reliant on the subgraph throughout our stack currently and if a revert happens, we don't want everyone else's experience to be impacted until we notice the issue and push a change.

Reply:
Yea, subgraph currently is built such way that there is no graceful way of warning about things. But Vincent do you experienced some address on some network actually causing the issue (once, or persistently)?

Reply:
no, I don't think I've ever experienced the subgraph going down because of a revert ever since we moved it into maintenance mode-I am not sure in what case a view function would cause a revert if there isn't any require/revert/assert statements in it.

Reply:
There is an experimental feature for non-fatal errors: https://thegraph.com/docs/en/developer/create-subgraph-hosted/#non-fatal-errors